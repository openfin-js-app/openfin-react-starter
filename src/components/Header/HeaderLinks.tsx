import * as React from 'react';
import { MouseEventHandler } from "react";
import cx from 'classnames';

import { WithStyles, withStyles } from '@material-ui/core';
import Button from '@material-ui/core/Button';

import AllOutIcon from '@material-ui/icons/AllOut';
import Remove from '@material-ui/icons/Remove';
import CropDin from '@material-ui/icons/CropDin';
import Clear from '@material-ui/icons/Clear';
import AspectRatio from '@material-ui/icons/AspectRatio';
import OpenInNew from '@material-ui/icons/OpenInNew';

import { headerLinksCompStyle as style } from '../../assets/jss/openfin-starter'

interface IProps extends WithStyles<typeof style> {
    windowsState:string,
    docked?:boolean,
    onUndock?:MouseEventHandler<any>,
    onSwitchToLaunchBar?:MouseEventHandler<any>,
    onMinimize:MouseEventHandler<any>,
    onMaximize:MouseEventHandler<any>,
    onClose:MouseEventHandler<any>,
}

class HeaderLinksComp extends React.Component<IProps,{}>{

    render(){

        const {
            classes, windowsState, docked,
            onSwitchToLaunchBar,
            onUndock,
            onMinimize, onMaximize, onClose
        } = this.props;

        return (<React.Fragment>
            {
                onSwitchToLaunchBar?
                    <Button className={cx(classes.menuBtn,classes.info)}
                            variant="fab"
                            mini
                            color="inherit"
                            aria-label="Switch to launch bar"
                            onClick={onSwitchToLaunchBar}
                    >
                        <OpenInNew/>
                    </Button>:null
            }
            {
                docked && onUndock ?
                    <Button className={cx(classes.menuBtn,classes.rose)}
                            variant="fab"
                            mini
                            color="inherit"
                            aria-label="Undock from the group"
                            onClick={onUndock}
                    >
                        <AllOutIcon/>
                    </Button>:null
            }
            <Button className={cx(classes.menuBtn,classes.success)}
                    variant="fab"
                    mini
                    color="inherit"
                    aria-label="Minimize"
                    onClick={onMinimize}
            >
                <Remove/>
            </Button>
            <Button className={cx(classes.menuBtn,classes.warning)}
                    variant="fab"
                    mini
                    color="inherit"
                    aria-label="Maximize"
                    onClick={onMaximize}
            >
                {windowsState==='normal'?<CropDin/>:<AspectRatio/>}
            </Button>
            <Button className={cx(classes.menuBtn,classes.danger)}
                    variant="fab"
                    mini
                    color="inherit"
                    aria-label="Close"
                    onClick={onClose}
            >
                <Clear/>
            </Button>
        </React.Fragment>);
    }

}

export default withStyles(style)(HeaderLinksComp);