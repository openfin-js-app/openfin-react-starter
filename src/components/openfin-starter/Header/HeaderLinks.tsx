import * as React from 'react';
import { MouseEventHandler } from "react";
import cx from 'classnames';

import { makeStyles } from '@material-ui/styles';
import Fab from '@material-ui/core/Fab';

import AllOutIcon from '@material-ui/icons/AllOut';
import Remove from '@material-ui/icons/Remove';
import CropDin from '@material-ui/icons/CropDin';
import Clear from '@material-ui/icons/Clear';
import AspectRatio from '@material-ui/icons/AspectRatio';
import OpenInNew from '@material-ui/icons/OpenInNew';

import { headerLinksCompStyle as style } from '../../../assets/jss/openfin-starter'

interface IProps {
    windowsState:string,
    docked?:boolean,
    onUndock?:MouseEventHandler<any>,
    onSwitchToLaunchBar?:MouseEventHandler<any>,
    onMinimize:MouseEventHandler<any>,
    onMaximize:MouseEventHandler<any>,
    onClose:MouseEventHandler<any>,
}

const useStyles = makeStyles(style);

const HeaderLinksComp:React.FunctionComponent<IProps> = (
    {
        windowsState, docked,
        onSwitchToLaunchBar,
        onUndock,
        onMinimize, onMaximize, onClose
    }
)=>{

    const classes = useStyles();

    return (<React.Fragment>
        {
            onSwitchToLaunchBar?
                <Fab className={cx(classes.menuBtn,classes.info)}
                     color="inherit"
                     aria-label="Switch to launch bar"
                     onClick={onSwitchToLaunchBar}
                >
                    <OpenInNew/>
                </Fab>:null
        }
        {
            docked && onUndock ?
                <Fab className={cx(classes.menuBtn,classes.rose)}
                     color="inherit"
                     aria-label="Undock from the group"
                     onClick={onUndock}
                >
                    <AllOutIcon/>
                </Fab>:null
        }
        <Fab className={cx(classes.menuBtn,classes.success)}
             color="inherit"
             aria-label="Minimize"
             onClick={onMinimize}
        >
            <Remove/>
        </Fab>
        <Fab className={cx(classes.menuBtn,classes.warning)}
             color="inherit"
             aria-label="Maximize"
             onClick={onMaximize}
        >
            {windowsState==='normal'?<CropDin/>:<AspectRatio/>}
        </Fab>
        <Fab className={cx(classes.menuBtn,classes.danger)}
             color="inherit"
             aria-label="Close"
             onClick={onClose}
        >
            <Clear/>
        </Fab>
    </React.Fragment>);

}

export default HeaderLinksComp;