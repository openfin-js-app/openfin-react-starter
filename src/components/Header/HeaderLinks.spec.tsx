import * as React from 'react';
import toJson from 'enzyme-to-json';
import { createShallow, createMount } from '@material-ui/core/test-utils';

import HeaderLinks from './HeaderLinks';

describe('HeaderLinks comp',()=>{

    let shallow;
    let mount;

    beforeAll(() => {
        mount = createMount();
    });

    afterAll(() => {
        mount.cleanUp();
    });


    beforeEach(() => {
        shallow = createShallow();
    });

    it('renders correctly by default',()=>{

        const wrapper = shallow(<HeaderLinks
            windowState={'normal'}
            onSwitchToLaunchBar={jest.fn()}
            onMinimize={jest.fn()}
            onMaximize={jest.fn()}
            onClose={jest.fn()}
        />);

        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();

    })

});