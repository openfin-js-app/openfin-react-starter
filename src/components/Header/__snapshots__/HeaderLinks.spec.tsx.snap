// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLinks comp renders correctly by default 1`] = `
<Fragment>
  <WithStyles(Button)
    aria-label="Switch to launch bar"
    className="HeaderLinksComp-menuBtn-1 HeaderLinksComp-info-2"
    color="inherit"
    mini={true}
    onClick={[MockFunction]}
    variant="fab"
  >
    <pure(OpenInNewIcon) />
  </WithStyles(Button)>
  <WithStyles(Button)
    aria-label="Minimize"
    className="HeaderLinksComp-menuBtn-1 HeaderLinksComp-success-3"
    color="inherit"
    mini={true}
    onClick={[MockFunction]}
    variant="fab"
  >
    <pure(RemoveIcon) />
  </WithStyles(Button)>
  <WithStyles(Button)
    aria-label="Maximize"
    className="HeaderLinksComp-menuBtn-1 HeaderLinksComp-warning-4"
    color="inherit"
    mini={true}
    onClick={[MockFunction]}
    variant="fab"
  >
    <pure(CropDinIcon) />
  </WithStyles(Button)>
  <WithStyles(Button)
    aria-label="Close"
    className="HeaderLinksComp-menuBtn-1 HeaderLinksComp-danger-5"
    color="inherit"
    mini={true}
    onClick={[MockFunction]}
    variant="fab"
  >
    <pure(ClearIcon) />
  </WithStyles(Button)>
</Fragment>
`;
